File: tests/conformance/fixtures/go/02-capabilities.go
package main

import sykli "github.com/yairfalse/sykli/sdk/go"

func main() {
	p := sykli.New()

	p.Task("compile").Run("go build -o /out/app ./cmd/server").
		Provides("binary", "/out/app")

	p.Task("migrate").Run("dbmate up").
		Provides("db-ready", "")

	p.Task("integration-test").Run("go test -tags=integration ./...").
		Needs("binary", "db-ready").Timeout(300)

	p.Emit()
}

File: tests/conformance/fixtures/elixir/02-capabilities.exs
use Sykli

pipeline do
  task "compile" do
    run "go build -o /out/app ./cmd/server"
    provides "binary", "/out/app"
  end

  task "migrate" do
    run "dbmate up"
    provides "db-ready"
  end

  task "integration-test" do
    run "go test -tags=integration ./..."
    needs ["binary", "db-ready"]
    timeout 300
  end
end
|> Sykli.Emitter.validate!()
|> Sykli.Emitter.to_json()
|> IO.puts()
